test/functional/framework.go

```diff
@@ -0,0 +1,147 @@
+package functional
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/ViaQ/logerr/log"
+	"github.com/openshift/cluster-logging-operator/internal/pkg/generator/forwarder"
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/pkg/certificates"
+	"github.com/openshift/cluster-logging-operator/pkg/constants"
+	"github.com/openshift/cluster-logging-operator/pkg/utils"
+	"github.com/openshift/cluster-logging-operator/test/client"
+	"github.com/openshift/cluster-logging-operator/test/helpers/oc"
+	"github.com/openshift/cluster-logging-operator/test/runtime"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	maxDuration time.Duration
+)
+
+//FluentddFunctionalFramework deploys stand alone fluentd with the fluent.conf as generated by input ClusterLogForwarder CR
+type FluentdFunctionalFramework struct {
+	Name      string
+	Namespace string
+	Conf      string
+	image     string
+	labels    map[string]string
+	Forwarder *logging.ClusterLogForwarder
+	test      *client.Test
+	pod       *corev1.Pod
+}
+
+func init() {
+	maxDuration, _ = time.ParseDuration("2m")
+}
+
+func NewFluentdFunctionalFramework() *FluentdFunctionalFramework {
+	verbosity := uint8(9)
+	if level, found := os.LookupEnv("LOG_LEVEL"); found {
+		if i, err := strconv.Atoi(level); err == nil {
+			verbosity = uint8(i)
+		}
+	}
+
+	opt := log.WithVerbosity(verbosity)
+	log.MustInitWithOptions("fluent-ftf", []log.Option{opt})
+	t := client.NewTest()
+	testName := fmt.Sprintf("test-fluent-%d", rand.Intn(1000))
+	framework := &FluentdFunctionalFramework{
+		Name:      testName,
+		Namespace: t.NS.Name,
+		image:     utils.GetComponentImage(constants.FluentdName),
+		labels: map[string]string{
+			"testtype": "functional",
+			"testname": testName,
+		},
+		test:      t,
+		Forwarder: runtime.NewClusterLogForwarder(),
+	}
+	return framework
+}
+
+func (f *FluentdFunctionalFramework) Cleanup() {
+	f.test.Close()
+}
+
+func (f *FluentdFunctionalFramework) RunCommand(cmdString string) (string, error) {
+	cmd := strings.Split(cmdString, " ")
```

Comment:
@jcantrill this is not doing good split for the command line parameters ... 
Can you change to import::

shellquote "github.com/kballard/go-shellquote"

cmd, _ := shellquote.Split(cmdString)

This splits the string much better and allows the exec command to really act like in shell

Reply:
Actually I'd prefer RunCommand(args...string) so that you have explicit control of the arguments in Go. Shell quoting and splitting on whitespace are both accident prone when you combine them with Go's string literal rules. e.g. what does this do: RunCommand("echo \"\\\"foo bar\\\"\"")?